onnxruntime/test/perftest/ort_test_session.cc

```diff
@@ -372,10 +432,12 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
           ORT_THROW("[ERROR] [TensorRT] The value for the key 'trt_cuda_graph_enable' should be a boolean i.e. true or false. Default value is false.\n");
         }
       } else {
-        ORT_THROW("[ERROR] [TensorRT] wrong key type entered. Choose from the following runtime key options that are available for TensorRT. ['device_id', 'trt_max_partition_iterations', 'trt_min_subgraph_size', 'trt_max_workspace_size', 'trt_fp16_enable', 'trt_int8_enable', 'trt_int8_calibration_table_name', 'trt_int8_use_native_calibration_table', 'trt_dla_enable', 'trt_dla_core', 'trt_dump_subgraphs', 'trt_engine_cache_enable', 'trt_engine_cache_path', 'trt_engine_decryption_enable', 'trt_engine_decryption_lib_path', 'trt_force_sequential_engine_build', 'trt_context_memory_sharing_enable', 'trt_layer_norm_fp32_fallback', 'trt_timing_cache_enable', 'trt_force_timing_cache', 'trt_detailed_build_log', 'trt_build_heuristics_enable', 'trt_sparsity_enable', 'trt_builder_optimization_level', 'trt_auxiliary_streams', 'trt_tactic_sources', 'trt_extra_plugin_lib_paths', 'trt_profile_min_shapes', 'trt_profile_max_shapes', 'trt_profile_opt_shapes', 'trt_cuda_graph_enable'] \n");
+        auto key_p = key.c_str();
+        auto value_p = value.c_str();
+        Ort::ThrowOnError(
+            api.UpdateTensorRTProviderOptions(&tensorrt_options, &key_p, &value_p, 1));
```

Comment:
Here might be a problem.
api.UpdateTensorRTProviderOptions() will call [TensorrtExecutionProviderInfo::UpdateProviderOptions()](https://github.com/gedoensmax/onnxruntime/blob/trt_timing_cache_path/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider_info.cc#L208) which it will reset the OrtTensorRTProviderOptionsV2 instance first and then update it with provided options. 
It means that only the last api.UpdateTensorRTProviderOptions() will be effective if you call multiple api.UpdateTensorRTProviderOptions(). 
I think we should change the behavior of this API to not reset the OrtTensorRTProviderOptionsV2 at the first place.

Reply:
Oh wow I did not notice that update always rewrites the full struct... In that case there is also an error in #17200 where I do the same for the perf test. I have to say that I dislike the "Update" terminology in that case.

Reply:
Totally agree and i will have another PR for changing this logic.

Reply:
That's great. For now I will first make sure to correct this in the onnxruntime perf test for TRT and CUDA. I will send a change tomorrow.

Reply:
If it is easy enough, maybe we can query the provider struct (`GetCUDAProviderOptionsAsString()`) and ensure that the user provided option is reflected in it or atleast print out the string